apps/opik-frontend/src/components/shared/SyntaxHighlighter/SyntaxHighlighter.tsx

```diff
@@ -2,68 +2,153 @@ import React, { useMemo, useState } from "react";
+  pretty = "pretty",
+}
 
-const OPTIONS = [
-  { value: CODE_TYPE_YAML, label: "YAML" },
-  { value: CODE_TYPE_JSON, label: "JSON" },
+const DEFAULT_OPTIONS = [
+  { value: MODE_TYPE.yaml, label: "YAML" },
+  { value: MODE_TYPE.json, label: "JSON" },
 ];
 
+const EXTENSION_MAP: { [key in MODE_TYPE]: LRLanguage | null } = {
+  [MODE_TYPE.yaml]: yamlLanguage,
+  [MODE_TYPE.json]: jsonLanguage,
+  [MODE_TYPE.pretty]: null,
+};
+
+type PrettifyConfig = {
+  fieldType: "input" | "output";
+};
+
 type SyntaxHighlighterProps = {
   data: object;
+  prettifyConfig?: PrettifyConfig;
 };
 
 const SyntaxHighlighter: React.FunctionComponent<SyntaxHighlighterProps> = ({
   data,
+  prettifyConfig,
 }) => {
-  const [codeLanguage, setCodeLanguage] = useState(CODE_TYPE_YAML);
+  const [mode, setMode] = useState(
+    prettifyConfig ? MODE_TYPE.pretty : MODE_TYPE.yaml,
+  );
   const theme = useCodemirrorTheme();
 
-  const formattedCodeByLanguage = useMemo(() => {
-    return codeLanguage == CODE_TYPE_YAML
-      ? yml.dump(data, { lineWidth: -1 }).trim()
-      : JSON.stringify(data, null, 2);
-  }, [codeLanguage, data]);
+  const code: {
+    message: string;
+    mode: MODE_TYPE;
+    prettified: boolean;
+    canBePrettified: boolean;
+  } = useMemo(() => {
+    const response = prettifyConfig
+      ? prettifyMessage(data, {
+          type: prettifyConfig.fieldType,
+        })
+      : {
+          message: data,
+          prettified: false,
+        };
 
-  const languageExtension =
-    codeLanguage === CODE_TYPE_JSON ? jsonLanguage : yamlLanguage;
+    switch (mode) {
+      case MODE_TYPE.yaml:
+        return {
+          message: yml.dump(data, { lineWidth: -1 }).trim(),
+          mode: MODE_TYPE.yaml,
+          prettified: false,
+          canBePrettified: response.prettified,
+        };
+      case MODE_TYPE.json:
+        return {
+          message: JSON.stringify(data, null, 2),
+          mode: MODE_TYPE.json,
+          prettified: false,
+          canBePrettified: response.prettified,
+        };
+      case MODE_TYPE.pretty:
+        return {
+          message: response.prettified
+            ? (response.message as string)
+            : yml.dump(data, { lineWidth: -1 }).trim(),
+          mode: response.prettified ? MODE_TYPE.pretty : MODE_TYPE.yaml,
+          prettified: response.prettified,
+          canBePrettified: response.prettified,
+        };
+
+      default:
+        return {
+          message: yml.dump({}, { lineWidth: -1 }).trim(),
+          mode: MODE_TYPE.yaml,
+          prettified: false,
+          canBePrettified: false,
+        };
+    }
+  }, [mode, data, prettifyConfig]);
+
+  const options = useMemo(() => {
+    if (prettifyConfig) {
+      return [
+        {
+          value: MODE_TYPE.pretty,
+          label: "Pretty ✨",
+          ...(!code.canBePrettified && {
+            disabled: !code.canBePrettified,
+            tooltip: "Pretty ✨ is not available yet for this trace format.",
```

Comment:
i don't think it is only about trace (spans are also possible?)
`Pretty ✨ is not available yet for this format` looks better in this component as it is a shared one